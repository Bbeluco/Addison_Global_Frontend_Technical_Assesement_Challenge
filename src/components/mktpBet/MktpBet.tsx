import React, { memo } from "react";
import { Button, StyleSheet, Text, TouchableOpacity, View } from 'react-native';

interface MktpBetType {
    setOptionSelected: (s: string) => void
}

const MktpBet = memo(function MktpBet({ setOptionSelected }: MktpBetType): React.JSX.Element {
    return  (
        <View style={styles.viewMktpBetOption}>
            <View style={styles.match}>
                <Text style={styles.matchInfo}>Man United</Text>
                <Text style={styles.matchInfo}>VS</Text>
                <Text style={styles.matchInfo}>Chelsea</Text>
            </View>
            <View style={styles.viewArea}>
                <Text style={styles.titleBetName}>To Win</Text>
                <View style={styles.bets}>
                    <TouchableOpacity style={[styles.viewArea, styles.btnChoseBet]}>
                        <Text>Man United</Text>
                        <Text>1.2</Text>
                    </TouchableOpacity>
                    <TouchableOpacity style={[styles.viewArea, styles.btnChoseBet]}>
                        <Text>Chelsea</Text>
                        <Text>2.2</Text>
                    </TouchableOpacity>
                </View>
            </View>
        </View>
    )
})

const styles = StyleSheet.create({
    viewMktpBetOption: {
        padding: '10%'
    },
    viewArea: {
        borderWidth: 1,
        borderColor: 'black', 
    },
    match: {
        borderWidth: 1,
        borderColor: 'black',
        flexDirection: 'row',
        justifyContent: 'space-between',
        padding: '10%'
    },
    matchInfo: {
        fontWeight: 'bold'
    },
    bets: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    btnChoseBet: {
        borderRadius: 10,
        alignItems: 'center',
        width: 100,
        padding: 5,
        margin: 5
    },
    titleBetName: {
        padding: 7
    }
})

export default MktpBet;