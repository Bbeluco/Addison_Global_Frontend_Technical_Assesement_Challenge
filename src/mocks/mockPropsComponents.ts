import { SEPARATOR } from "../constants/selectOptionConstants"
import mockReturnApi from "./mockReturnApi"

export const betOptionMock = [{ 'option': `button${SEPARATOR}0${SEPARATOR}0${SEPARATOR}0`, 'teamName': 'b', 'titleBet': 'c', 'price': 1 }]

export const pushSelectedOptionInfoMock = {
    add(option: string, teamName: string, titleBet: string, price: number) {},
    remove(itemRemove: number) {},
}

export const navigator = {
    onClose() {},
}

// export function pushSelectedOptionInfoMock(option: string, teamName: string, titleBet: string, price: number) {}
export const titleBetMock = 'Bet123'
export const selectionBetNameMock = 'To win'
export const selectionPriceMock = 12
export const eventIndexMock = 0
export const marketIndexMock = 0
export const selectionIndexMock = 0
export const marketNameMock = 'Market name'
export const selectionsMock = mockReturnApi[0]['markets'][0]['selections']
