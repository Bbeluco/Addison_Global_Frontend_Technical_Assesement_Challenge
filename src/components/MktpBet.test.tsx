import MktpBet from "./mktpBet/MktpBet";
import { betOptionMock, pushSelectedOptionInfoMock } from "../mocks/mockPropsComponents";
import mockReturnApi, { mockReturnEmptyMarket } from "../mocks/mockReturnApi";
import { render, screen } from "@testing-library/react-native";

describe('<MktpBet />', () => {
    it('should render the component correctly', () => {
        render(
            <MktpBet
                betOption={betOptionMock}
                pushSelectedOptionInfo={pushSelectedOptionInfoMock.add}
                apiResponseBetAvaible={mockReturnApi}
            />
        )
        expect(screen.toJSON()).toMatchSnapshot();
    })

    it('should not render when market is empty', () => {
        render(
            <MktpBet
                betOption={betOptionMock}
                pushSelectedOptionInfo={pushSelectedOptionInfoMock.add}
                apiResponseBetAvaible={mockReturnEmptyMarket}
            />
        )

        expect(screen.queryByTestId('betting_options')).toBeNull()
    })
})