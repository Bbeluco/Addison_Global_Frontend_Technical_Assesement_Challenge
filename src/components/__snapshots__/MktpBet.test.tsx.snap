// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MktpBet /> should render the component correctly 1`] = `
<View
  data-testID="mktp_bet"
  style={
    {
      "margin": 5,
    }
  }
>
  <View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "10%",
        }
      }
    >
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        Real Madrid
      </Text>
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        VS
      </Text>
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        Barcelona
      </Text>
    </View>
    <View>
      <View
        data-testID="betting_options"
        style={
          {
            "borderColor": "black",
            "borderWidth": 1,
          }
        }
      >
        <Text
          style={
            {
              "padding": 7,
            }
          }
        >
          Team to Win
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "green",
                  "borderColor": "black",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 5,
                  "width": 100,
                }
              }
              testID="button_Real_Madrid_betOption"
            >
              <Text>
                Real Madrid
              </Text>
              <Text>
                1.23
              </Text>
            </View>
          </View>
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "black",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 5,
                  "width": 100,
                }
              }
              testID="button_Barcelona_betOption"
            >
              <Text>
                Barcelona
              </Text>
              <Text>
                2.05
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View
        data-testID="betting_options"
        style={
          {
            "borderColor": "black",
            "borderWidth": 1,
          }
        }
      >
        <Text
          style={
            {
              "padding": 7,
            }
          }
        >
          Player to Score First
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "black",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 5,
                  "width": 100,
                }
              }
              testID="button_Ronaldo_betOption"
            >
              <Text>
                Ronaldo
              </Text>
              <Text>
                1.15
              </Text>
            </View>
          </View>
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "black",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 5,
                  "width": 100,
                }
              }
              testID="button_Messi_betOption"
            >
              <Text>
                Messi
              </Text>
              <Text>
                1.3
              </Text>
            </View>
          </View>
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "black",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 5,
                  "width": 100,
                }
              }
              testID="button_Bale_betOption"
            >
              <Text>
                Bale
              </Text>
              <Text>
                1.35
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "10%",
        }
      }
    >
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        Atletico Madrid
      </Text>
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        VS
      </Text>
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        Malaga
      </Text>
    </View>
    <View>
      <View
        data-testID="betting_options"
        style={
          {
            "borderColor": "black",
            "borderWidth": 1,
          }
        }
      >
        <Text
          style={
            {
              "padding": 7,
            }
          }
        >
          Team to Win
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "black",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 5,
                  "width": 100,
                }
              }
              testID="button_Atletico_betOption"
            >
              <Text>
                Atletico
              </Text>
              <Text>
                1.4
              </Text>
            </View>
          </View>
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "black",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 5,
                  "width": 100,
                }
              }
              testID="button_Malaga_betOption"
            >
              <Text>
                Malaga
              </Text>
              <Text>
                3.05
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "10%",
        }
      }
    >
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        Empty Event that shouldn't render
      </Text>
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        VS
      </Text>
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      />
    </View>
  </View>
</View>
`;
